### DISCUSSION TITLE
Running on Cloudflare Workers

### DISCUSSION THREAD

**Discussion by sorenbs** [rhashimoto/wa-sqlite#100] [Q&A]
Has anybody experimented with running wa-sqlite on Cloudflare Workers?

I am interested in using the recently announced [Cloud Backed SQLite](https://sqlite.org/cloudsqlite/doc/trunk/www/index.wiki) VFS module (or implement a similar mechanism in Javascript) and run it in a Cloudflare Durable Object.

**Comment by rhashimoto** [comment]
I'm not aware of anything in that specific area, but there are examples of using an in-browser SQLite VFS to access a remotely stored database:

* @phiresky wrote a VFS for SQL.js to [access a database file using HTTP Range](https://phiresky.github.io/blog/2021/hosting-sqlite-databases-on-github-pages/) from a static web site. This work inspired a lot of similar efforts; I'm sure I'm not aware of all of them.
* @ydylla wrote a [WebTorrent VFS](https://github.com/rhashimoto/wa-sqlite/pull/21).
* Here's another [WebTorrent VFS](https://boredcaveman.xyz/post/0x2_static-torrent-website-p2p-queries.html).
* @LoneRifle wrote a [writable Amazon S3 VFS](https://github.com/LoneRifle/wa-sqlite-s3vfs). This is the closest effort I know of to your question.

In addition, there is development on syncing distributed local databases (including in-browser) to be "eventually consistent". [cr-sqlite](https://github.com/vlcn-io/cr-sqlite) is one example using (a fork of) wa-sqlite.

**Reply by sorenbs** [reply]
These are some great resources - thank you! I think my first experiment is going to be to get one of these wasm builds targeting the browser to work on Cloudflare Workers.
